src/Bicep.Cli/Program.cs

```diff
@@ -100,25 +100,38 @@ private ILoggerFactory CreateLoggerFactory()
         private int Build(ILogger logger, BuildArguments arguments)
         {
             var diagnosticLogger = new BicepDiagnosticLogger(logger);
-            var bicepPaths = arguments.Files.Select(f => PathHelper.ResolvePath(f)).ToArray();
+            var bicepPath = PathHelper.ResolvePath(arguments.InputFile);
+
             if (arguments.OutputToStdOut)
             {
-                BuildManyFilesToStdOut(diagnosticLogger, bicepPaths);
+                BuildToStdout(diagnosticLogger, bicepPath);
             }
-            else
+            else if (arguments.OutputDir is not null)
             {
-                foreach (string bicepPath in bicepPaths)
+                var outputDir = PathHelper.ResolvePath(arguments.OutputDir);
+                if (!Directory.Exists(outputDir))
                 {
-                    string outputPath = PathHelper.GetDefaultOutputPath(bicepPath);
-                    BuildSingleFile(diagnosticLogger, bicepPath, outputPath);
+                    throw new CommandLineException($"The specified output directory \"{outputDir}\" does not exist.");
```

Comment:
I thought that we would create the output dir if it doesn't exist

Reply:
We could, but it didn't feel necessary - most of the time I imagine the directory will be precreated, and if not, it should take a trivial amount of scripting to create it.

By not implementing it, we avoid having to worry about edge cases like the user mistyping a path and having unexpected directories created.